external/kafka/src/main/scala/org/apache/spark/streaming/kafka/NewKafkaCluster.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.kafka
+
+import java.util
+import java.util.Collections
+
+import scala.collection.JavaConverters._
+import scala.reflect._
+
+import org.apache.kafka.clients.consumer.{KafkaConsumer, OffsetAndMetadata, OffsetResetStrategy}
+import org.apache.kafka.common.{PartitionInfo, TopicPartition}
+
+import org.apache.spark.SparkException
+
+/**
+ * @param kafkaParams Kafka <a href="http://kafka.apache.org/documentation.html#configuration">
+ *                    configuration parameters</a>.
+ *                    Requires "bootstrap.servers" to be set with Kafka broker(s),
+ *                    NOT zookeeper servers, specified in host1:port1,host2:port2 form
+ */
+private[spark]
+class NewKafkaCluster[K: ClassTag, V: ClassTag](val kafkaParams: Map[String, String])
```

Comment:
This is another place where we can't use 'new' in the class name . See https://github.com/apache/spark/pull/9007 which recently got merged. We have to make this a public class because of valid use-cases like https://issues.apache.org/jira/browse/SPARK-13106